### PR TITLE
Add distributed key generation with RockX

### PR DESCRIPTION
**Pull request by shanejearley**
Remaining:
- [x] Document package and CLI usage
- [x] Mock key generation API service for apps/contract development
- [x] Add `reshareValidator` to @casimir/keys service
- [ ] Update `addValidator` and `registerValidator` usage to latest SSV contracts
- [ ] ~~Mock key generation API service routes for @casimir/keys tests~~

I'm going to put off the last item (mocking network calls) for the sake of getting to full integration faster, and since @casimir/keys does not do much else at this point anyways (so there is not much else to test besides the keygen input and output). I'm also going to come back to a few things on this service after an intermediate PR, including:
- Resharing test and live scenario
- Add ceremony proof (assure DKG ceremony was valid)

### CODE REVIEW

File: services/keys/scripts/dev.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    await $`npx esno -r dotenv/config src/index.ts help`
+    const dkgServiceUrl = 'http://0.0.0.0:8000'
+    const groups = [[1, 2, 3, 4], [1, 2, 3, 4]]
+    for (const group of groups) {  
+        console.log(`Starting ceremony for operators: ${group.join(',')}`)
+        await $`npx esno -r dotenv/config src/index.ts create-validator --dkgServiceUrl ${dkgServiceUrl} --operatorIds ${group.join(',')}`
+        console.log('Completed ceremony...')
+
+        if (group !== groups[groups.length - 1]) {
+            console.log('Waiting for 5 seconds before starting the next ceremony...')
```
Comment (ccali11) [inline services/keys/scripts/dev.ts:13]: CEREMONY?! Is that the technical term??

Comment (ccali11) [reply services/keys/scripts/dev.ts:13]: Cool term if so.

File: services/keys/src/providers/dkg.ts [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+export class DKG {
+    /** Key generation service URL */
+    serviceUrl: string
+
+    constructor(options: DKGOptions) {
+        this.serviceUrl = options.serviceUrl
+    }
+
+    /**
+     * Start a new key generation ceremony
```
Comment (ccali11) [inline services/keys/src/providers/dkg.ts:17]: And a beautiful ceremony it was...

Comment (ccali11) [reply services/keys/src/providers/dkg.ts:17]: This all looks great.

**Review by ccali11 — APPROVED** [review]
Looks great, man.

File: README.md [hunk 3]
```diff
@@ -119,6 +119,14 @@ Deploy a contract, specifically [contracts/ethereum/src/SSVManager.sol](contract
 npm run deploy:ssv --workspace @casimir/ethereum
 ```
 
+### Keys
+
+Generate new validator keys with [@casimir/keys](services/keys/).
+
+```zsh
+npm run compose --workspace @casimir/keys
```
[the remark this answers is not in the document]
Comment (ccali11) [reply README.md:127]: @shanejearley - My only relevant comment in this PR^

[the remark this answers is not in the document]
Comment (shanejearley) [reply README.md:127]: Good catch, should have been removed.

**Pull request merged**